Logging of negative shotnoise values is done incorrectly -> Exceptions

Logging is done incorrectly when pixels are negative, see

https://github.com/AstarVienna/ScopeSim/blob/84e269863b17fac2b6c9d465f0c7ecaf3840ab0f/scopesim/effects/electronic.py#L351-L353

The signature of the `warning` function is `warning(msg, *args, **kwargs)`[¶](https://docs.python.org/3/library/logging.html#logging.Logger.warning), so the second two arguments are used to format the first argument.

It is hard to create an example that is easy to reproduce and also simple. However the problem seems to occur more easily when running code through `pytest`. Partial trace:
```
../.conda/envs/micadowise/lib/python3.9/site-packages/scopesim/optics/optical_train.py:305: in readout  
    hdul = detector_array.readout(self.image_planes, array_effects,                     
../.conda/envs/micadowise/lib/python3.9/site-packages/scopesim/detector/detector_array.py:80: in readout
    detector = effect.apply_to(detector)
../.conda/envs/micadowise/lib/python3.9/site-packages/scopesim/effects/electronic.py:352: in apply_to
    logging.warning("Effect ShotNoise:", np.sum(data < 0),
...
../.conda/envs/micadowise/lib/python3.9/logging/__init__.py:663: in format
    record.message = record.getMessage()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <LogRecord: root, 30, /home/micado/.conda/envs/micadowise/lib/python3.9/site-packages/scopesim/ef
fects/electronic.py, 352, "Effect ShotNoise:">      

    def getMessage(self):
        """
        Return the message for this LogRecord.
     
        Return the message for this LogRecord after merging any user-supplied
        arguments with the message.
        """
        msg = str(self.msg)
        if self.args:
>           msg = msg % self.args
E           TypeError: not all arguments converted during string formatting

../.conda/envs/micadowise/lib/python3.9/logging/__init__.py:367: TypeError
```

Suggestion is to use fstrings:
```python
     logging.warning(f"Effect ShotNoise: {np.sum(data < 0)} negative pixels") 
```

--- a/scopesim/effects/electronic.py
+++ b/scopesim/effects/electronic.py
@@ -348,9 +348,11 @@ class ShotNoise(Effect):
             # basically the same. For large arrays the normal distribution
             # takes only 60% as long as the poisson distribution
             data = det._hdu.data
-            if np.any(data > 0):
-                logging.warning("Effect ShotNoise:", np.sum(data < 0),
-                                "negative pixels")
+
+            # Check if there are negative values in the data
+            negvals = np.sum(data < 0)
+            if negvals:
+                logging.warning(f"Effect ShotNoise: {negvals} negative pixels")
 
             below = data < 2**20
             above = np.invert(below)

